src/System.Net.Security/tests/FunctionalTests/SslStreamNegotiatedCipherSuiteTest.cs

```diff
@@ -0,0 +1,357 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.Net.Test.Common;
+using System.Security.Authentication;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Security.Tests
+{
+    using Configuration = System.Net.Test.Common.Configuration;
+
+    public class NegotiatedCipherSuiteTest
+    {
+        private static bool IsKnownPlatformSupportingTls13 => PlatformDetection.IsUbuntu1810OrHigher;
+        private static bool Tls13Supported { get; set; } = ProtocolsSupported(SslProtocols.Tls13);
+
+        private static HashSet<TlsCipherSuite> _tls13CipherSuiteLookup = new HashSet<TlsCipherSuite>(GetTls13CipherSuites());
+        private static HashSet<TlsCipherSuite> _tls12CipherSuiteLookup = new HashSet<TlsCipherSuite>(GetTls12CipherSuites());
+        private static HashSet<TlsCipherSuite> _tls10And11CipherSuiteLookup = new HashSet<TlsCipherSuite>(GetTls10And11CipherSuites());
+
+        private static Dictionary<SslProtocols, HashSet<TlsCipherSuite>> _protocolCipherSuiteLookup = new Dictionary<SslProtocols, HashSet<TlsCipherSuite>>()
+        {
+            { SslProtocols.Tls12, _tls12CipherSuiteLookup },
+            { SslProtocols.Tls11, _tls10And11CipherSuiteLookup },
+            { SslProtocols.Tls, _tls10And11CipherSuiteLookup },
+        };
+
+        [ConditionalFact(nameof(IsKnownPlatformSupportingTls13))]
+        public void Tls13IsSupportedFlagWorksCorrectly()
```

Comment:
Please consider the current naming format for tests across all of System.Net.*.

We use a three-part naming format.

See: https://github.com/dotnet/corefx/blob/c3effbbd54289597f711cceda6bc3a56e825230b/src/System.Net.Security/tests/FunctionalTests/SslAuthenticationOptionsTest.cs#L20